pkg/controller/servicemesh/controlplane/reconciler.go

```diff
@@ -422,37 +422,89 @@ func (r *controlPlaneInstanceReconciler) applyDisconnectedSettings(ctx context.C
 	if err != nil {
 		return smcpSpec, err
 	}
+
+	// these need to exist or we'll get errors trying to set the images
+	if smcpSpec.Istio == nil {
+		smcpSpec.Istio = make(v1.HelmValuesType)
+	}
+	if smcpSpec.ThreeScale == nil {
+		smcpSpec.ThreeScale = make(v1.HelmValuesType)
+	}
 	switch version {
 	case maistra.V1_0, maistra.UndefinedVersion:
-		r.updateImageField(smcpSpec.Istio, "security.image", common.Config.OLM.Images.V1_0.Citadel)
-		r.updateImageField(smcpSpec.Istio, "galley.image", common.Config.OLM.Images.V1_0.Galley)
-		r.updateImageField(smcpSpec.Istio, "grafana.image", common.Config.OLM.Images.V1_0.Grafana)
-		r.updateImageField(smcpSpec.Istio, "mixer.image", common.Config.OLM.Images.V1_0.Mixer)
-		r.updateImageField(smcpSpec.Istio, "pilot.image", common.Config.OLM.Images.V1_0.Pilot)
-		r.updateImageField(smcpSpec.Istio, "prometheus.image", common.Config.OLM.Images.V1_0.Prometheus)
-		r.updateImageField(smcpSpec.Istio, "global.proxy_init.image", common.Config.OLM.Images.V1_0.ProxyInit)
-		r.updateImageField(smcpSpec.Istio, "global.proxy.image", common.Config.OLM.Images.V1_0.ProxyV2)
-		r.updateImageField(smcpSpec.Istio, "sidecarInjectorWebhook.image", common.Config.OLM.Images.V1_0.SidecarInjector)
-		r.updateImageField(smcpSpec.ThreeScale, "image", common.Config.OLM.Images.V1_0.ThreeScale)
+		if err := r.updateImageField(smcpSpec.Istio, "security.image", common.Config.OLM.Images.V1_0.Citadel); err != nil {
+			return smcpSpec, err
+		}
+		if err := r.updateImageField(smcpSpec.Istio, "galley.image", common.Config.OLM.Images.V1_0.Galley); err != nil {
+			return smcpSpec, err
+		}
+		if err := r.updateImageField(smcpSpec.Istio, "grafana.image", common.Config.OLM.Images.V1_0.Grafana); err != nil {
+			return smcpSpec, err
+		}
+		if err := r.updateImageField(smcpSpec.Istio, "mixer.image", common.Config.OLM.Images.V1_0.Mixer); err != nil {
+			return smcpSpec, err
+		}
+		if err := r.updateImageField(smcpSpec.Istio, "pilot.image", common.Config.OLM.Images.V1_0.Pilot); err != nil {
+			return smcpSpec, err
+		}
+		if err := r.updateImageField(smcpSpec.Istio, "prometheus.image", common.Config.OLM.Images.V1_0.Prometheus); err != nil {
+			return smcpSpec, err
+		}
+		if err := r.updateImageField(smcpSpec.Istio, "global.proxy_init.image", common.Config.OLM.Images.V1_0.ProxyInit); err != nil {
+			return smcpSpec, err
+		}
+		if err := r.updateImageField(smcpSpec.Istio, "global.proxy.image", common.Config.OLM.Images.V1_0.ProxyV2); err != nil {
+			return smcpSpec, err
+		}
+		if err := r.updateImageField(smcpSpec.Istio, "sidecarInjectorWebhook.image", common.Config.OLM.Images.V1_0.SidecarInjector); err != nil {
+			return smcpSpec, err
+		}
+		if err := r.updateImageField(smcpSpec.ThreeScale, "image", common.Config.OLM.Images.V1_0.ThreeScale); err != nil {
+			return smcpSpec, err
+		}
```

Comment:
I'd love to see this replaced with a for-loop and then extracted into a separate function, but ...